StantUiIosLibrary/Classes/Utils/ViewComponents/Gesturizer.swift

```diff
@@ -0,0 +1,53 @@
+//
+//  Gesturizer.swift
+//  StantUiIosLibrary
+//
+//  Created by Mac Mini Novo on 25/07/19.
+//
+
+import UIKit
+
+public class Gesturizer {
+    
+    private let viewController: UIViewController
+    private let backgroundView: UIView
+    private var initialTouchPoint: CGPoint = CGPoint(x: 0,y: 0)
+    
+    init(viewController: UIViewController,
+         backgroundView: UIView) {
+        self.viewController = viewController
+        self.backgroundView = backgroundView
+    }
+    
+    func swipeDownGesture(sender: UIGestureRecognizer) {
```

Comment:
This function has a lot of information and conditionals. Let's try to extract those responsibilities, avoiding many conditionals too. For example, what does means when the state is changed? What steps are they doing inside this block, for both conditionals? 

We can try to apply an extraction for responsibilities in methods or classes to make the code cleaner and readable.

Reply:
I refactored it. Thanks.